### ISSUE TITLE
[Promotions] Not longer eligible for promotion at checkout-complete with "Order total" rule

### ISSUE THREAD

**Issue by psihius** [Sylius/Sylius#7836]
⚠️ 🛣 **This issue is a part of the** [Sylius Public Roadmap](https://github.com/orgs/Sylius/projects/2) 🛣 ⚠️

* Create a promotion coupon, configure it as "Order total" X sum and "Order discount" 50%
* Add a products, so the total cart sum is just over "X" (or at least when discount is applied, the resulting sum is bellow "X").
* Apply the coupon in the cart
* Proceed with the checkout
* When pressing "Complete order" and "complete" step - get an error "You are no longer eligible for this promotion Y"

After debugging the code, I found out that "ItemTotalRuleChecker" is checking against Order::getItemsTotal that is returning an already adjusted sum, not the item original cost, that results in the order being no longer eligible for the promotion.

System should check against original sum for items, not an adjusted one when validating eligibility.

**Comment by blankcheck** [comment]
I can confirm this - sadly it happened to us in production on really busy cyber Monday.

Without time to fully investigate, we applied hot fix (in case someone needs it):

Override Order model's getPromotionSubjectTotal() method

```
    public function getPromotionSubjectTotal()
    {
    		$itemsTotalWithoutDiscounts = 0;
    		
    		foreach ($this->items as $item) {
    			$itemsTotalWithoutDiscounts += $item->getUnitPrice() * $item->getQuantity(); 
    		}
    		
    		return $itemsTotalWithoutDiscounts;
    }
```

**Comment by gabiudrescu** [comment]
I confirm I can reproduce this on demo.sylius.com

![demo sylius com_en_us_checkout_complete](https://user-images.githubusercontent.com/5156054/41641755-fffe68ee-746e-11e8-98bb-d0bdeae63323.png)
![demo sylius com_admin_promotions_20984_edit](https://user-images.githubusercontent.com/5156054/41641757-01de8b4e-746f-11e8-9a84-7f727bf88424.png)

**Comment by lchrusciel** [comment]
@gabiudrescu Are you willing to write a behat to reproduce this problem?

**Comment by blankcheck** [comment]
@rafaelrocha007 yes

**Comment by pamil** [comment]
Fixed by #9512.
